components/bl-totop-component/src/index.js

```diff
@@ -0,0 +1,54 @@
+import { useRef } from "react";
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const Transform = {
+  MAIN: "50%",
+  X: "translateX(-50%)",
+  Y: "translateY(-50%)",
+};
+
+export default function TotopComponent({ component }) {
+  const { classList, offset, position, backgroundColor, color, size, iconSize, indent = "20px", element } = component;
+
+  const styles = {
+    backgroundColor,
+    width: size,
+    height: size,
+    top: position.includes("top") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    bottom: position.includes("bottom") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    right: position.includes("right") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    left: position.includes("left") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    transform: position.includes("-center") ? Transform.X : position.includes("center-") ? Transform.Y : undefined,
+  };
+
+  const iconStyles = {
+    width: iconSize,
+    height: iconSize,
+    fill: color,
+  };
+
+  const handleScroll = () => {
+    window.scrollTo({ top: (element?.el?.offsetTop || 0) + offset, behavior: "smooth" });
+  };
+
+  const totopRef = useRef();
+
+  component.el = totopRef.current;
+
+  return (
+    <div ref={totopRef} className={cn("bl-customComponent-totop", classList)} style={styles} onClick={handleScroll}>
+      <TotopIcon iconStyles={iconStyles} />
```

Comment:
TotopIcon already have icon in name, so no need to name props as "iconStyles", use just "styles", and pass iconStyles for it